src/slider/tp-slider.ts

```diff
@@ -91,7 +98,126 @@ export class TPSliderElement extends HTMLElement {
 	 */
 	static get observedAttributes(): string[] {
 		// Observed attributes.
-		return [ 'current-slide', 'flexible-height', 'infinite', 'swipe', 'per-view', 'step' ];
+		return [ 'current-slide', 'flexible-height', 'infinite', 'swipe', 'per-view', 'step', 'draggable' ];
+	}
+
+	/**
+	 * Check and initialize draggable behavior.
+	 */
+	checkDraggable() {
+		// Check if draggable attribute is set to "yes".
+		if ( 'yes' === this.getAttribute( 'draggable' ) ) {
+			this.initializeDraggableEvents();
+		}
+	}
+
+	/**
+	 * Initialize draggable event listeners.
+	 */
+	initializeDraggableEvents() {
+		// Add event listeners to handle dragging behavior (start, move, stop) for the slider.
+		this.addEventListener( 'mousedown', this.handleMouseDown.bind( this ) );
+		this.addEventListener( 'mousemove', this.handleMouseMove.bind( this ) );
+		this.addEventListener( 'mouseup', this.handleMouseUp.bind( this ) );
+		this.addEventListener( 'mouseleave', this.handleMouseUp.bind( this ) );
+
+		// Add cursor changes for grab on hover
+		this.addEventListener( 'mouseover', this.handleMouseOver.bind( this ) );
+		this.addEventListener( 'mouseout', this.handleMouseOut.bind( this ) );
+	}
+
+	/**
+	 * Mouse down event - Start dragging.
+	 *
+	 * @param {MouseEvent} event - The mouse event object.
+	 */
+	handleMouseDown( event: MouseEvent ) {
+		// Set dragging state to true
+		this.isDragging = true;
+
+		// Store the initial X position when the mouse is pressed
+		this.dragStartX = event.clientX;
+
+		// Prevent the default behavior (e.g., text selection)
+		event.preventDefault();
+
+		// Update styles for grabbing and disable text selection.
+		this.style.cursor = 'grabbing';
```

Comment:
@AvinashVinod03 Perhaps this should be attributes on the component itself. Please avoid using CSS in JS

Reply:
@junaidbhura do you want that we want to use classList.add and give styles with the css file itself ?